tests/unit/amo/components/TestAddonReviewListItem.js

```diff
@@ -188,6 +189,49 @@ describe(__filename, () => {
     );
   });
 
+  it('renders a delete link for a user review', () => {
+    const review = signInAndDispatchSavedReview();
+    const root = render({ review });
+
+    const deleteLink = renderControls(root).find('.AddonReviewListItem-delete');
+    expect(deleteLink).toHaveLength(1);
+    expect(deleteLink.children()).toHaveText('Delete my review');
+  });
+
+  it('does not render delete link when review belongs to another user', () => {
+    const review = signInAndDispatchSavedReview({
+      siteUserId: 123,
+      reviewUserId: 987,
+    });
+    const root = render({ review });
+
+    expect(
+      renderControls(root).find('.AddonReviewListItem-delete'),
+    ).toHaveLength(0);
+  });
+
+  it('dispatches deleteReview when a user clicks the delete link', () => {
+    const review = signInAndDispatchSavedReview();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const root = render({ review });
+    const { errorHandler } = root.instance().props;
+
+    const deleteButton = renderControls(root).find(
+      '.AddonReviewListItem-delete',
+    );
+    const clickEvent = createFakeEvent();
+    deleteButton.simulate('click', clickEvent);
+
+    sinon.assert.called(clickEvent.preventDefault);
+    sinon.assert.calledWith(
+      dispatchSpy,
+      deleteAddonReview({
+        errorHandlerId: errorHandler.id,
+        reviewId: review.id,
+      }),
+    );
```

Comment:
should you show that this link has length 0 now (after using the delete link) ~

.find('.AddonReviewListItem-delete').toHaveLength(0)

Reply:
This is an interesting idea, and might have been simple before changing this to use `ConfirmButton`, but now the test does not actually click on the button, it just calls the `onConfirm` method of the button. In order to do this now we'd have to first click on the `ConfirmButton` link and then click on the actual "Confirm" button that is displayed by the `ConfirmButton` component. That seems like way too much for this test. 

What do you think, @kumar303? Can you think of a simpler way of confirming that the "Delete" link is no longer present immediately after the deletion occurs?

Reply:
https://github.com/mozilla/addons-frontend/pull/6084/files#r213419834

> should you show that this link has length 0 now (after using the delete link)

No, because that's the responsibility of `ConfirmButton`.

However, I don't see a test case for this! Good catch. 

Can you add one to https://github.com/mozilla/addons-frontend/blob/master/tests/unit/ui/components/TestConfirmButton.js ?

Reply:
I think we're talking about two different things here. I believe what @rebmullin was suggesting is a test that shows that after deleting the review/reply the edit link no longer appears, but maybe not because the entire review/reply would disappear so the edit link must also disappear.

I think what you're referring to @kumar303 is a test that shows that when you click the link for a `ConfirmButton`, that, in addition to the confirmation dialog appearing, the link also disappears. Is that correct?